buildSrc/src/main/kotlin/org/jetbrains/conventions/base-java.gradle.kts

```diff
@@ -0,0 +1,22 @@
+package org.jetbrains.conventions
+
+/**
+ * Base configuration for Java projects.
+ *
+ * This convention plugin contains shared Java config for both the [KotlinJvmPlugin] convention plugin and
+ * the Gradle Plugin subproject (which cannot have the `kotlin("jvm")` plugin applied).
+ */
+
+plugins {
+    `java`
+}
+
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(8))
+    }
+}
```

Comment:
If we apply `toolchain` for all modules, they'll run tests on Java 8 instead of the JDK matrix on CI, we need to apply more toolchains in `Test` which will slow  down the build, I believe we should not use `toolchain` for now. Check

- https://github.com/detekt/detekt/pull/5735
- https://github.com/Kotlin/dokka/blob/f5581c51d0de23d2b038e00b1a152faebaef0920/.github/workflows/gradle-test.pr.yml#L12-L13

Reply:
Good catch!

Toolchains can still help with this though, because it's possible to [set them per-task](https://docs.gradle.org/current/userguide/toolchains.html#toolchains_for_tasks). I did something similar for MockK - I'll put together a PR.

Reply:
Ah right, yes that would be slower. I agree, using GitHub parallel workflows is best, not multiple Gradle tasks.